base image builds are failing

The error log ends in this:

```
Step #3: [0m[91m[740 / 822] Compiling coverage.cc; 4s processwrapper-sandbox ... (29 actions, 28 running)
Step #3: [0m[91mERROR: /src/centipede/BUILD:502:8: Compiling blob_file_test.cc failed: (Exit 1): clang-15 failed: error executing command /usr/local/bin/clang-15 -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer -g0 -O2 ... (remaining 53 arguments skipped)
Step #3: 
Step #3: Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
Step #3: [0m[91mblob_file_test.cc:50:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(appender->Open(path));
Step #3:     ^
Step #3: blob_file_test.cc:51:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(appender->Append(input1));
Step #3:     ^
Step #3: blob_file_test.cc:52:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(appender->Append(input2));
Step #3:     ^
Step #3: blob_file_test.cc:53:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(appender->Close());
Step #3:     ^
Step #3: blob_file_test.cc:59:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(reader->Open(path));
Step #3:     ^
Step #3: blob_file_test.cc:60:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(reader->Read(blob));
Step #3:     ^
Step #3: blob_file_test.cc:62:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(reader->Read(blob));
Step #3:     ^
Step #3: blob_file_test.cc:65:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(reader->Close());
Step #3:     ^
Step #3: blob_file_test.cc:71:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(appender->Open(path));
Step #3:     ^
Step #3: blob_file_test.cc:72:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(appender->Append(input3));
Step #3:     ^
Step #3: blob_file_test.cc:73:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(appender->Close());
Step #3:     ^
Step #3: blob_file_test.cc:79:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(reader->Open(path));
Step #3:     ^
Step #3: blob_file_test.cc:80:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(reader->Read(blob));
Step #3:     ^
Step #3: blob_file_test.cc:82:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(reader->Read(blob));
Step #3:     ^
Step #3: blob_file_test.cc:84:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(reader->Read(blob));
Step #3:     ^
Step #3: blob_file_test.cc:87:5: error: use of undeclared identifier 'EXPECT_OK'
Step #3:     EXPECT_OK(reader->Close());
Step #3:     ^
Step #3: blob_file_test.cc:117:3: error: use of undeclared identifier 'EXPECT_OK'
Step #3:   EXPECT_OK(appender->Open(path));
Step #3:   ^
Step #3: blob_file_test.cc:120:3: error: use of undeclared identifier 'EXPECT_OK'
Step #3:   EXPECT_OK(appender->Close());
Step #3:   ^
Step #3: blob_file_test.cc:130:3: error: use of undeclared identifier 'EXPECT_OK'
Step #3:   EXPECT_OK(reader->Open(path));
Step #3:   ^
Step #3: fatal error: too many errors emitted, stopping now [-ferror-limit=]
Step #3: 20 errors generated.
Step #3: [0m[91mINFO: Elapsed time: 45.305s, Critical Path: 14.59s
Step #3: INFO: 800 processes: 382 internal, 418 processwrapper-sandbox.
Step #3: [0m[91mFAILED: Build did NOT complete successfully
Step #3: [0m[91mFAILED: Build did NOT complete successfully
Step #3: The command '/bin/sh -c precompile_centipede' returned a non-zero code: 1
Finished Step #3
ERROR
ERROR: build step 3 "gcr.io/cloud-builders/docker" failed: step exited with non-zero status: 1
```

--- a/infra/base-images/base-builder/Dockerfile
+++ b/infra/base-images/base-builder/Dockerfile
@@ -124,7 +124,9 @@ RUN cd $SRC && \
     rm -rf examples $SRC/oss-fuzz.tar.gz
 
 RUN cd $SRC && \
-    git clone --depth 1 https://github.com/google/centipede.git
+    git clone https://github.com/google/centipede.git && \
+    cd centipede && \
+    git checkout de14f117d01fd5a54d0c3129503420d9d6f23c3f
 
 # Do precompiles before copying other scripts for better cache efficiency.
 COPY precompile_afl /usr/local/bin/

--- a/infra/base-images/base-builder/Dockerfile
+++ b/infra/base-images/base-builder/Dockerfile
@@ -116,6 +116,10 @@ RUN git clone https://github.com/AFLplusplus/AFLplusplus.git aflplusplus && \
     wget --no-check-certificate -O oss.sh https://raw.githubusercontent.com/vanhauser-thc/binary_blobs/master/oss.sh && \
     chmod 755 oss.sh
 
+# Do precompiles before copying other scripts for better cache efficiency.
+COPY precompile_afl /usr/local/bin/
+RUN precompile_afl
+
 RUN cd $SRC && \
     curl -L -O https://github.com/google/honggfuzz/archive/oss-fuzz.tar.gz && \
     mkdir honggfuzz && \
@@ -123,17 +127,14 @@ RUN cd $SRC && \
     tar -xzv --strip-components=1 -f $SRC/oss-fuzz.tar.gz && \
     rm -rf examples $SRC/oss-fuzz.tar.gz
 
-RUN cd $SRC && \
-    git clone https://github.com/google/centipede.git && \
-    cd centipede && \
-    git checkout de14f117d01fd5a54d0c3129503420d9d6f23c3f
-
-# Do precompiles before copying other scripts for better cache efficiency.
-COPY precompile_afl /usr/local/bin/
-RUN precompile_afl
 
 COPY precompile_honggfuzz /usr/local/bin/
 RUN precompile_honggfuzz
+
+RUN cd $SRC && \
+    git clone https://github.com/google/centipede.git && \
+    cd centipede && \
+    git checkout a976295b2acaff8461a906c461f5bb25dafdf557
 
 COPY precompile_centipede /usr/local/bin/
 RUN precompile_centipede

--- a/infra/base-images/base-builder/precompile_centipede
+++ b/infra/base-images/base-builder/precompile_centipede
@@ -32,4 +32,8 @@ bazel --bazelrc=/tmp/centipede.bazelrc build -c opt :all
 /clang/bin/clang++ "$SRC/centipede/weak_sancov_stubs.cc" -c -o \
   "$SRC/centipede/weak.o"
 
+echo 'Removing extra stuff leftover to avoid bloating image.'
+
+rm -rf /clang-*.tgz /clang /root/.cache/*
+
 echo 'Done.'

